From 92a75475af230321d4b7ba83679d0b034c37a360 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Tue, 30 Sep 2025 15:19:14 +0200
Subject: [PATCH 250/553] QSysInfo: don't pass QStringLiteral to
 QWinRegistryKey::stringValue()

The stringValue() function took QStringView from the initial commit of
the class, so passing QStringLiteral (which creates and destroys a
full QString instance, and, until recently, even injected an
additional isNull() branch when converted to QStringView) was never
necessary; a UTF-16 string literal would have always sufficed, and be
faster.

Since the API will will soon be changed to take const wchar_t*, use a
wchar_t literal instead of char16_t one. For QStringView, it doesn't
matter (on Windows, at least), but it keeps the call compatible with
the future QWinRegistryKey API.

Amends 80996d2e5d6d04b08452cf507f6ced7d8a09068c.

Pick-to: 6.8 6.5
Change-Id: If88fb074c72bf3a41adf5cfb4b68f185c24481fd
Reviewed-by: Friedemann Kleint <Friedemann.Kleint@qt.io>
(cherry picked from commit bbc8f8f7a8a4d8fe15871e428bec6c49abf32673)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/global/qsysinfo.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/corelib/global/qsysinfo.cpp b/src/corelib/global/qsysinfo.cpp
index e5bba170ed4..78e50cd8580 100644
--- a/src/corelib/global/qsysinfo.cpp
+++ b/src/corelib/global/qsysinfo.cpp
@@ -1059,7 +1059,7 @@ QByteArray QSysInfo::machineUniqueId()
 #elif defined(Q_OS_WIN)
     // Let's poke at the registry
     const QString machineGuid = QWinRegistryKey(HKEY_LOCAL_MACHINE, LR"(SOFTWARE\Microsoft\Cryptography)")
-                                .stringValue(u"MachineGuid"_s);
+                                .stringValue(L"MachineGuid");
     if (!machineGuid.isEmpty())
         return machineGuid.toLatin1();
 #endif
-- 
2.51.2

